projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -157,6 +158,13 @@ public function set_site_notices_menu_class( array $menu ) {
 		return $menu;
 	}
 
+	/**
+	 * Adds Inbox menu.
+	 */
+	public function add_inbox_menu() {
+		add_menu_page( __( 'Inbox', 'jetpack' ), __( 'Inbox', 'jetpack' ), 'edit_posts', 'https://wordpress.com/email/inbox/' . $this->domain, null, 'dashicons-email', '4.64424' );
+	}
+
```

Comment:
@AllTerrainDeveloper, I believe that this functionality should **not** be in this class, as that means that it will be visible for all Jetpack clients. I think we should be adding the new menu in `class-wpcom-admin-menu.php` and `class-atomic-admin-menu.php`.

I am also not clear on how we can limit who sees the new menu -- we want to be able to test this internally before we launch to all users. I think we should use a similar approach to #19394, where we handled this via a filter that defaults to `false`.

Reply:
Thank you @daledupreez, **_very useful_** information